packages/app/src/runner/SpecRunnerDropdown.vue

```diff
@@ -40,15 +53,27 @@
 </template>
 
 <script lang="ts" setup>
+import { ref, watch } from 'vue'
 import TransitionQuickFade from '@cy/components/transitions/TransitionQuickFade.vue'
 
 import { Popover, PopoverButton, PopoverPanel } from '@headlessui/vue'
 
 const props = withDefaults(defineProps<{
   variant?: 'panel'
   align?: 'left' | 'right'
+  disabled?: boolean
 }>(), {
   variant: undefined,
   align: 'right',
+  disabled: false,
 })
+
+const popoverKey = ref(false)
+
+// Watch the disabled prop to ensure that any open popovers are closed
+// if the dropdown button is subsequently disabled
+watch(() => props.disabled, () => {
+  popoverKey.value = !popoverKey.value
+})
```

Comment:
So I'm cheating a bit. I want to update the key when the disabled prop changes so that the popover remounts, closing any open popover in the process. It seems as though the component doesn't expose a close function except through its v-slots(?), which we don't have a way of accessing directly from outside the template.

I could generate a unique key every time it changes, by incrementing a counter or using `Lodash _.uniqueId()`. But I don't necessarily need a *unique* key, I just need one that is different from the current key. So I flip back and forth through the boolean state as necessary and interpolate the value into a string (key cannot be a boolean type, so I stringify it).

But you raise a good point; I could just interpolate the disabled prop value directly and ditch the watcher entirely!

Reply:
Good instinct here @rachelruderman - I did a bit of a double take when I saw this, heh.